worker/remoterelations/remoterelations.go

```diff
@@ -240,23 +252,45 @@ func (w *Worker) handleApplicationChanges(applicationIds []string) error {
 
 	for i, result := range results {
 		name := applicationIds[i]
-		if result.Error != nil {
-			// The the remote application has been removed, stop its worker.
-			if params.IsCodeNotFound(result.Error) {
-				if err := w.runner.StopWorker(name); err != nil {
-					return err
+
+		// The remote application may refer to an offer that has been removed from
+		// the offering model, or it may refer to a new offer with a different UUID.
+		// If it is for a new offer, we need to stop any current worker for the old offer.
+		appGone := result.Error != nil && params.IsCodeNotFound(result.Error)
+		if result.Error != nil && !appGone {
+			return errors.Annotatef(result.Error, "querying remote application %q", name)
```

Comment:
Should this not be `saas application`?

Reply:
Not here as this worker runs on both consume and offer models and "remote application" is the generic term